src/path.c

```diff
@@ -236,6 +236,146 @@ void twin_path_arc(twin_path_t *path,
     twin_path_set_matrix(path, save);
 }
 
+static twin_angle_t vector_angle(twin_fixed_t ux,
+                                 twin_fixed_t uy,
+                                 twin_fixed_t vx,
+                                 twin_fixed_t vy)
+{
+    twin_fixed_t dot = twin_fixed_mul(ux, vx) + twin_fixed_mul(uy, vy);
+
+    twin_fixed_t ua =
+        twin_fixed_sqrt(twin_fixed_mul(ux, ux) + twin_fixed_mul(uy, uy));
+    twin_fixed_t va =
+        twin_fixed_sqrt(twin_fixed_mul(vx, vx) + twin_fixed_mul(vy, vy));
+
+    /* cos(theta) = dot / (|u| * |v|) */
+    twin_fixed_t cos_theta = twin_fixed_div(dot, twin_fixed_mul(ua, va));
+    twin_fixed_t cross = twin_fixed_mul(ux, vy) - twin_fixed_mul(uy, vx);
+    twin_angle_t angle = twin_acos(cos_theta);
+    return (cross < 0) ? -angle : angle;
+}
+
+typedef struct {
+    twin_fixed_t cx, cy;
+    twin_angle_t start, extent;
+} twin_ellipse_para_t;
```

Comment:
If you are abbreviating "parameter," use "param" instead of "para," which is an existing noun. See https://www.merriam-webster.com/dictionary/para